tests/test_00_unit/test_00_index_reader/__init__.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+
+from ccp import index_reader
+
+
+class TestProject(unittest.TestCase):
+    """
+    Test the Project class representing an
+    object in container pipeline
+    """
+
+    def setUp(self):
+        self.entry = {
+            "id": "1",
+            "app-id": "foo",
+            "job-id": "bar",
+            "desired-tag": "latest",
+            "git-branch": "master",
+            "git-url": "https://github.com/someurl",
+            "git-path": "foo/",
+            "target-file": "Dockerfile",
+            "build-context": "./",
+            "notify-email": "abc@example/com",
+            "depends-on": ["centos/centos:latest", "centos/centos:7"],
+            "prebuild-script": "hooks/script.sh",
+            "prebuild-context": "/",
+        }
+        self.namespace = "ccp"
+
+    def test_load_entries(self):
+        """
+        Tests creating Project class object which loads
+        entries
+        """
+        index_reader.Project(self.entry, self.namespace)
+
+    def test_string_representation(self):
+        """
+        Test __str__ representation of Project object
+        """
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(
+            "foo-bar-latest",
+            project.__str__()
+        )
+
+    def test_replace_dot_slash_colon_(self):
+        """
+        Test the helper method to replace . / : _ with hyphen
+        """
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(
+            "a-b-c-d",
+            project.replace_dot_slash_colon_("a.b/c_d")
+        )
+
+    def test_depends_on(self):
+        """
+        Tests depends_on field processing and defaults
+        """
+        # test the default value
+        project = index_reader.Project(self.entry, self.namespace)
+        # test if multiple/list of depends-on is converted to string
+        self.assertIsInstance(
+            project.depends_on,
+            str
+        )
+
+        # test if commas are included with multiple values of depends_on
+        self.assertIn(",", project.depends_on)
+
+        # test the actual values
+        self.assertEqual(
+            "ccp-centos-centos-latest,ccp-centos-centos-7",
+            project.depends_on
+        )
+
+    def test_desired_tag(self):
+        """
+        Tests desired tag processing and defaults
+        """
+        # test a custom value
+        self.entry["desired-tag"] = "release"
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(project.desired_tag, "release")
+
+        # test empty value
+        self.entry["desired-tag"] = ""
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(project.desired_tag, "latest")
```

Comment:
can we test it with putting `desired-tag` to some int lets say `1`?

Reply:
@bamachrn 

Take a look at [Project](https://github.com/dharmit/ccp-openshift/blob/master/ccp/index_reader.py#L37) class in index_reader.py
```
class Project(object):
    """
    Class for storing and processing a container index project
    """

    def __init__(self, entry, namespace):
```
its initialization starts after loading entry from yaml using baseloader. It does not take any yaml entry.

And after loading yaml file using baseloader, all the data types are converted to string.

The unit tests are testing the unit functionalities, if we you provide the int value, it has to be provided in tests before loading the yaml files, the Project class deals with operations after yaml is loaded and converted to string (by baseLoader).

Having int values for desired-tag in test-index should be in scope of functional tests.

Reply:
:+1: